src/software/test_util/test_util.cpp

```diff
@@ -173,4 +212,64 @@ namespace Test
                    << "Value 1 was " << val1 << ", value 2 was " << val2;
         }
     }
+
+    ::testing::AssertionResult TestUtil::equalWithinTolerance(
+        const RobotState &state1, const RobotState &state2, const double linear_tolerance,
+        const Angle &angular_tolerance)
+    {
+        if (!TestUtil::equalWithinTolerance(state1.position(), state2.position(),
+                                            linear_tolerance))
+        {
+            return ::testing::AssertionFailure()
+                   << "The first state's position was " << state1.position()
+                   << ", the second state's position was " << state2.position();
```

Comment:
would be good for debug to print out all the differences between the states, not just the first one

Reply:
done